addons/mrp/models/mrp_production.py

```diff
@@ -583,6 +583,7 @@ def _workorders_create(self, bom, bom_data):
             })
             if workorders:
                 workorders[-1].next_work_order_id = workorder.id
+                workorders[-1]._init_nextworkorder_states()
```

Comment:
it's already done line 580

Reply:
No, l580 we give it an initial value (for all but the first, pending) but if there is a batch_size of 0, then we want to allow workorders to start in parallel. 
That's why we go over the state once again, but using the info from workorders[-1] to decide what its state should really be (although it's only really used in mrp_workorder...).

Reply:
ok you are right